src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -25,25 +28,12 @@ public class QnAService {
 
     @Transactional
     public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
+        LocalDateTime regDateTime = LocalDateTime.now();
         Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
+
+        question.delete(loginUser);
+        DeleteHistories deleteHistories = question.getDeleteHistories(regDateTime);
```

Comment:
```suggestion
        DeleteHistories deleteHistories = question.delete(loginUser);
```

위와 같이 삭제한 뒤 이력이 반환되는건 어떨까요?

```
요청자: 질문아 니가 가진 질문과 답변을 삭제해줘
질문: 그래, 삭제했어. 자, 니가 삭제 요청한 삭제 이력이야.
```

와 같이 이해할 수 있을것 같아요!